docs/performance/getting-started.md

```diff
@@ -0,0 +1,73 @@
+# Getting started with Performance Monitoring
+
+##  Basic usage
+
+**TBD** basic explainer
+
+```ts
+constructor(private afp: AngularFirePerformance, private afs: AngularFirestore) {}
+
+ngOnInit() {
+  this.articles = afs.collection('articles')
+      .collection('articles', ref => ref.orderBy('publishedAt', 'desc'))
+      .snapshotChanges()
+      .pipe(
+        this.afp.trace('getArticles'),
+        map(articles => ...)
+      );
+}
+```
+
+`trace(name:string)` will trace the time it takes for your observable to either complete or emit it's first value. Beyond the basic trace there are three other operators:
+
+### `traceComplete(name:string)`
+
+Traces the observable until the completion.
+
+### `traceUntil(name:string, test: (T) => Boolean)`
+
+Traces the observable until the first emission that passes the provided test.
+
+### `traceFirst(name: string)`
+
+Traces the observable until the first emission or the first emission that matches the provided test.
+
+## Advanced usage
+
+### `trace$(...) => Observable<void>`
+
+`(name:string)`
+`(name:string, options: TraceOptions)`
+
+Observable version of `firebase/perf`'s `.trace` method; the basis for `AngularFirePerfomance`'s pipes.
+
+`.subscribe()` is equivalent to calling `.start()`
+`.unsubscribe()` is equivalent to calling `.stop()`
+
+### `TraceOptions`
+
+**TBD explain how each option is used by `.trace$`**
```

Comment:
"Extend the amount of detail recorded with your trace by using the TraceOptions type to add custom metrics and attributes to your traces"

maybe put the TraceOptions type definition before the sample, and throw in comments after each field. Also, pluckMetrics and pluckAttributes don't seem to be in the actual [type definition]( https://github.com/angular/angularfire2/blob/2bff468e0025a84800034a9da297677372206f4a/src/performance/performance.ts#L8)

```ts
export type TraceOptions = {
  pluckMetrics?: [string],  // ??
  pluckAttributes?: [string], // ??
  metrics?: { [key:string]: number }, // add custom metrics to a trace
  attributes?: { [key:string]: string }, // add custom attributes to a trace
  attribute$?: { [key:string]: Observable<string> }, // observable that emits new values for custom attributes
 incrementMetric$: { [key:string]: Observable<number|void|null|undefined> }, // observable that emits new increments for custom metrics 
  metric$?: { [key:string]: Observable<number> }// observable that emits new values for custom attributes
};
```

Reply:
Seems I left pluck in a hanging commit, will rebase that in & think more on the options and naming here.

Reply:
Dropping pluck, having dug up my implementation I realize why I didn't include it. It's too complicated for now. Let's see how users adopt the implemented options before getting too ahead of ourselves.